### ISSUE TITLE
MCEdit Crashes on Launch after closing with the X button

### ISSUE THREAD

**Issue by pswiifanwmi** [Khroki/MCEdit-Unified#595]
I was working with MCEdit today, and I closed it with the X button, but now when I start it, it gives me an error that looks to be about languages, but I could be wrong.  I have deleted all of the files that I thought were associated with MCEdit too, but this still happens.
I tried this on another computer and the same thing happened after closing with the X button.  Both were running Windows 10 64bit.
I'm not an avid Python user.  This is the log file.  Were there instructions explicitly stating not to close with the button that I missed or ignored.  How do I fix this?

http://pastebin.com/pwq1TnfZ

**Comment by LaChal** [comment]
You are not exactly wrong. An there's no warning about not using the window close button.

MCEdit look for block and item names in Minecraft folders.
The error you have suggest the `%AppData%\.minecraft\ assets\indexes` folder does not contain the needed file. `%AppData%` is typically `C:\Users\<Your User Name>\AppData`.
Have look in this directory. There shall be a `legacy.json` file and some other ones like `1.8.json`.

Does MCEdit crashes?
Is there something else in the console which opens with MCEdit?

**Comment by pswiifanwmi** [comment]
![image](https://cloud.githubusercontent.com/assets/10968291/13464214/0be37aca-e05e-11e5-8ba3-ca0ea1122f91.png)

This is the command prompt I get when launching MCEdit.  I looked in the directory and there is a legacy.json file there.  I would classify this as a crash because the splash screen pops up for a couple seconds and closes to bring up this lovely command prompt.
The strange thing is that it works before I press the X button.  It does not work afterward and I have to move to another computer afterward and try again.
I should add that I'm working with both vanilla and modded Minecraft (Technic) saves.  It seems to work when I close with quit though, but it permanently breaks when I close with X.
My peer told me that in the mclanges.py file, line 88 is faulty or something because it thinks that a crash occurred (while there wasn't one just closing with X), but there was no log file to read the error off of or something.  I'm not sure though because I can't remember what he said

**Comment by Rubisk** [comment]
On phone, someone swap lines 78-80 with lines 81-82

**Comment by LaChal** [comment]
I honestly can't see the link between closing MCEdit with the window close button and the fact this `%AppData%\.minecraft\ assets\indexes` lack some files...

This folder shall contain the Minecraft resources needed to translate names. Each file in this directory shall be named after Minecraft versions you have on your computer. If you have installed 1.7 and 1.8.8, there shall be `1.7.json` and `1.8.8.json` files in `%AppData%\.minecraft\ assets\indexes`.
The line 88 in `mclangres.py` is faulty because the folder does not contain anything else than the `legacy.json` file.

The issue you're facing *may* be due to the modded version of the game you installed. It is possible that this mod moves the files...
It also *may* be related to some snapshot version of Minecraft...

I have a sort of fix for this issue (moving two lines in the code will make the job). But, I think it is better to look a bit deeper into this.

If you agree, you can make a simple test: create an empty `1.0.json` file in `%AppData%\.minecraft\ assets\indexes`. Just be sure the extension is only `.json`, not something like `.json.txt`.
After this, the expected behaviour of MCEdit is to run normally, but without block and item names translated. You may not notice this, except looking in the log file for `Could not find <something> resource name.`

In any case, please report here :smile:

@Podshot: that's what I saw first when tracking the bug origin :wink:

**Comment by pswiifanwmi** [comment]
I fixed the issue.  I had recently reinstalled Windows on both of these computers so the .minecraft folder was deleted.  I downloaded 1.3.2 yesterday when I was using the program, but I hadn't used 1.8 at all (or 1.7).  They weren't installed.  I reinstalled 1.8 and voilà, MCEdit launched.  I thank you guys for helping.  The one weird thing was that it launched before without anything in that folder since I hadn't played vanilla.  I'm not sure how the program works or anything, but how did it launch without a \indexes folder?  Anyway, the problem is solved by reinstalling Minecraft 1.8.8.  Thanks Again!
